library/std/src/sys/pal/uefi/time.rs

```diff
@@ -112,6 +123,52 @@ pub(crate) mod system_time_internal {
 
         Duration::new(utc_epoch, t.nanosecond)
     }
+
+    /// This algorithm is taken from: http://howardhinnant.github.io/date_algorithms.html
+    pub(crate) const fn uefi_time_from_duration(
+        dur: crate::time::Duration,
+        daylight: u8,
+        timezone: i16,
+    ) -> r_efi::system::Time {
+        const SECS_IN_MINUTE: u64 = 60;
+        const SECS_IN_HOUR: u64 = SECS_IN_MINUTE * 60;
+        const SECS_IN_DAY: u64 = SECS_IN_HOUR * 24;
+
+        let secs = if timezone == r_efi::efi::UNSPECIFIED_TIMEZONE {
+            dur.as_secs()
+        } else {
+            dur.as_secs().checked_add_signed(-timezone as i64).unwrap()
+        };
+        let days = secs / SECS_IN_DAY;
+        let remaining_secs = secs % SECS_IN_DAY;
+        let z = days + 719468;
+        let era = z / 146097;
+        let doe = z - (era * 146097);
```

Comment:
This is clear by operator precedence, and the source doesn't use parenthesis here either.
```suggestion
        let doe = z - era * 146097;
```